test/distributed/test_store.py

```diff
@@ -311,6 +315,28 @@ def test_multi_worker_with_fixed_world_size(self):
     def test_multi_worker_with_nonfixed_world_size(self):
         self._multi_worker_helper(None)
 
+class StoreBasedBarrierTest(MultiThreadedTestCase):
+    @property
+    def world_size(self):
+        return 1
```

Comment:
We're not really testing much here with a world_size of one :)

Reply:
Yeah, the only thing is that the vast majority of our tests use `FileStore`. So this would also be directly testing that `store_based_barrier` works with `HashStore` and `TcpStore`.

I was not able to make it larger than 1 for TCPStore when it was using `MultiThreadedTestCase`, so I need to update it. I was planning on splitting out `init_process_group` from that class